### ISSUE TITLE
When exceuting datashield.pkg_status() and datashield.pkg_version() it is not giving the expected output

### ISSUE THREAD

**Issue by sidohaakma** [molgenis/molgenis-r-datashield#46]
Run `datashield.pkg_status()` with DSOpal only.

```
$package_status
     package ninfea  moba  dnbc  elfe
1     dsBase   TRUE  TRUE  TRUE  TRUE
2 dsExposome   TRUE FALSE FALSE FALSE
3 dsBetaTest  FALSE FALSE FALSE  TRUE
5  resourcer   TRUE  TRUE  TRUE FALSE
$version_status
     package ninfea  moba  dnbc  elfe
1     dsBase  6.1.0 6.1.0 6.1.0 6.0.1
2 dsExposome  1.0.1  <NA>  <NA>  <NA>
3 dsBetaTest   <NA>  <NA>  <NA> 0.2.0
5  resourcer  1.0.1 1.0.1 1.0.1  <NA>
```

Run `datashield.pkg_status()` with DSOpal and DSMolgenisArmadillo combined only.

```
$package_status
            package alspac  eden  inma  moba ninfea
base           base   TRUE FALSE FALSE FALSE  FALSE
dsBase       dsBase   TRUE  TRUE  TRUE  TRUE   TRUE
stats         stats   TRUE FALSE FALSE FALSE  FALSE
1                 1  FALSE FALSE FALSE FALSE  FALSE
2                 2  FALSE FALSE FALSE FALSE  FALSE
resourcer resourcer   TRUE FALSE  TRUE  TRUE   TRUE
3                 3  FALSE FALSE FALSE FALSE  FALSE

$version_status
            package alspac eden inma moba ninfea
base           base   NULL   NA   NA   NA     NA
dsBase       dsBase  6.1.0    2    1    1      2
stats         stats   NULL   NA   NA   NA     NA
1                 1   NULL   NA   NA   NA     NA
2                 2   NULL   NA   NA   NA     NA
resourcer resourcer  1.1.0   NA    1    1      1
3                 3   NULL   NA   NA   NA     NA
```

**Comment by mswertz** [comment]
@StuartWeather is this an issue in datashield client?

**Comment by timcadman** [comment]
This is still not working and we should look at it again

**Comment by timcadman** [comment]
I've done some digging, and the issue is that the `DSMolgenisArmadillo` method `dsListMethods` returns a different structure than the opal method. Opal returns:

```
List of 1
 $ server1:'data.frame':	85 obs. of  6 variables:
  ..$ name   : chr [1:85] "NROW" "anderson.darling.testDS" "asFactorDS1" "asListDS" ...
  ..$ type   : chr [1:85] "aggregate" "aggregate" "aggregate" "aggregate" ...
  ..$ class  : chr [1:85] "function" "function" "function" "function" ...
  ..$ value  : chr [1:85] "base::NROW" "dsExposome::anderson.darling.testDS" "dsBase::asFactorDS1" "dsBase::asListDS" ...
  ..$ package: chr [1:85] "dsBase" "dsExposome" "dsBase" "dsBase" ...
  ..$ version: chr [1:85] "6.3.0" "2.0.9" "6.3.0" "6.3.0" ...
```

Whilst Armadillo returns a list of lists:
```
> str(res)
List of 1
 $ study1:'data.frame':	71 obs. of  6 variables:
  ..$ name   :List of 71
  .. ..$ : chr "extractQuantilesDS1"
  .. ..$ : chr "is.character"
  .. ..$ : chr "dataFrameSubsetDS1"
  .. ..$ : chr "meanSdGpDS"
  .. ..$ : chr "scatterPlotDS"
  .. ..$ : chr "setSeedDS"
  .. ..$ : chr "minMaxRandDS"
  .. ..$ : chr "matrixDetDS1"
  .. ..$ : chr "numNaDS"
  .. ..$ : chr "asListDS"
  .. ..$ : chr "is.numeric"
  .. ..$ : chr "gamlssDS"
  .. ..$ : chr "t.test"
  .. ..$ : chr "glmPredictDS.ag"
  .. ..$ : chr "extractQuantilesDS2"
  .. ..$ : chr "is.list"
  .. ..$ : chr "lengthDS"
  .. ..$ : chr "heatmapPlotDS"
  .. ..$ : chr "dimDS"
  .. ..$ : chr "is.null"
  .. ..$ : chr "glmSummaryDS.ag"
  .. ..$ : chr "meanDS"
  .. ..$ : chr "tableDS"
  .. ..$ : chr "is.factor"
  .. ..$ : chr "corDS"
  .. ..$ : chr "testObjExistsDS"
  .. ..$ : chr "hetcorDS"
  .. ..$ : chr "isValidDS"
  .. ..$ : chr "table1DDS"
  .. ..$ : chr "asFactorDS1"
  .. ..$ : chr "glmDS2"
  .. ..$ : chr "glmDS1"
  .. ..$ : chr "covDS"
  .. ..$ : chr "kurtosisDS2"
  .. ..$ : chr "kurtosisDS1"
  .. ..$ : chr "lmerSLMADS2"
  .. ..$ : chr "rmDS"
  .. ..$ : chr "exists"
  .. ..$ : chr "metadataDS"
  .. ..$ : chr "rangeDS"
  .. ..$ : chr "table2DDS"
  .. ..$ : chr "NROW"
  .. ..$ : chr "scoreVectDS"
  .. ..$ : chr "tapplyDS"
  .. ..$ : chr "levelsDS"
  .. ..$ : chr "densityGridDS"
  .. ..$ : chr "lexisDS1"
  .. ..$ : chr "histogramDS1"
  .. ..$ : chr "histogramDS2"
  .. ..$ : chr "tableDS2"
  .. ..$ : chr "checkNegValueDS"
  .. ..$ : chr "quantileMeanDS"
  .. ..$ : chr "ranksSecureDS3"
  .. ..$ : chr "ranksSecureDS1"
  .. ..$ : chr "varDS"
  .. ..$ : chr "boxPlotGGDS"
  .. ..$ : chr "colnamesDS"
  .. ..$ : chr "lsDS"
  .. ..$ : chr "classDS"
  .. ..$ : chr "skewnessDS2"
  .. ..$ : chr "isNaDS"
  .. ..$ : chr "listDisclosureSettingsDS"
  .. ..$ : chr "skewnessDS1"
  .. ..$ : chr "glmerSLMADS2"
  .. ..$ : chr "glmSLMADS1"
  .. ..$ : chr "glmSLMADS2"
  .. ..$ : chr "aucDS"
  .. ..$ : chr "namesDS"
  .. ..$ : chr "corTestDS"
  .. ..$ : chr "miceDS"
  .. ..$ : chr "messageDS"
  ..$ value  :List of 71
  .. ..$ : chr "dsBase::extractQuantilesDS1"
  .. ..$ : chr "base::is.character"
  .. ..$ : chr "dsBase::dataFrameSubsetDS1"
  .. ..$ : chr "dsBase::meanSdGpDS"
  .. ..$ : chr "dsBase::scatterPlotDS"
  .. ..$ : chr "dsBase::setSeedDS"
  .. ..$ : chr "dsBase::minMaxRandDS"
  .. ..$ : chr "dsBase::matrixDetDS1"
  .. ..$ : chr "dsBase::numNaDS"
  .. ..$ : chr "dsBase::asListDS"
  .. ..$ : chr "base::is.numeric"
  .. ..$ : chr "dsBase::gamlssDS"
  .. ..$ : chr "stats::t.test"
  .. ..$ : chr "dsBase::glmPredictDS.ag"
  .. ..$ : chr "dsBase::extractQuantilesDS2"
  .. ..$ : chr "base::is.list"
  .. ..$ : chr "dsBase::lengthDS"
  .. ..$ : chr "dsBase::heatmapPlotDS"
  .. ..$ : chr "dsBase::dimDS"
  .. ..$ : chr "base::is.null"
  .. ..$ : chr "dsBase::glmSummaryDS.ag"
  .. ..$ : chr "dsBase::meanDS"
  .. ..$ : chr "dsBase::tableDS"
  .. ..$ : chr "base::is.factor"
  .. ..$ : chr "dsBase::corDS"
  .. ..$ : chr "dsBase::testObjExistsDS"
  .. ..$ : chr "dsBase::hetcorDS"
  .. ..$ : chr "dsBase::isValidDS"
  .. ..$ : chr "dsBase::table1DDS"
  .. ..$ : chr "dsBase::asFactorDS1"
  .. ..$ : chr "dsBase::glmDS2"
  .. ..$ : chr "dsBase::glmDS1"
  .. ..$ : chr "dsBase::covDS"
  .. ..$ : chr "dsBase::kurtosisDS2"
  .. ..$ : chr "dsBase::kurtosisDS1"
  .. ..$ : chr "dsBase::lmerSLMADS2"
  .. ..$ : chr "dsBase::rmDS"
  .. ..$ : chr "base::exists"
  .. ..$ : chr "dsBase::metadataDS"
  .. ..$ : chr "dsBase::rangeDS"
  .. ..$ : chr "dsBase::table2DDS"
  .. ..$ : chr "base::NROW"
  .. ..$ : chr "dsBase::scoreVectDS"
  .. ..$ : chr "dsBase::tapplyDS"
  .. ..$ : chr "dsBase::levelsDS"
  .. ..$ : chr "dsBase::densityGridDS"
  .. ..$ : chr "dsBase::lexisDS1"
  .. ..$ : chr "dsBase::histogramDS1"
  .. ..$ : chr "dsBase::histogramDS2"
  .. ..$ : chr "dsBase::tableDS2"
  .. ..$ : chr "dsBase::checkNegValueDS"
  .. ..$ : chr "dsBase::quantileMeanDS"
  .. ..$ : chr "dsBase::ranksSecureDS3"
  .. ..$ : chr "dsBase::ranksSecureDS1"
  .. ..$ : chr "dsBase::varDS"
  .. ..$ : chr "dsBase::boxPlotGGDS"
  .. ..$ : chr "dsBase::colnamesDS"
  .. ..$ : chr "dsBase::lsDS"
  .. ..$ : chr "dsBase::classDS"
  .. ..$ : chr "dsBase::skewnessDS2"
  .. ..$ : chr "dsBase::isNaDS"
  .. ..$ : chr "dsBase::listDisclosureSettingsDS"
  .. ..$ : chr "dsBase::skewnessDS1"
  .. ..$ : chr "dsBase::glmerSLMADS2"
  .. ..$ : chr "dsBase::glmSLMADS1"
  .. ..$ : chr "dsBase::glmSLMADS2"
  .. ..$ : chr "dsBase::aucDS"
  .. ..$ : chr "dsBase::namesDS"
  .. ..$ : chr "dsBase::corTestDS"
  .. ..$ : chr "dsBase::miceDS"
  .. ..$ : chr "dsBase::messageDS"
  ..$ version:List of 71
  .. ..$ : chr "6.3.0"
  .. ..$ : NULL
  .. ..$ : chr "6.3.0"
  .. ..$ : chr "6.3.0"
  .. ..$ : chr "6.3.0"
  .. ..$ : chr "6.3.0"
  .. ..$ : chr "6.3.0"
  .. ..$ : chr "6.3.0"
  .. ..$ : chr "6.3.0"
  .. ..$ : chr "6.3.0"
  .. ..$ : NULL
  .. ..$ : chr "6.3.0"
  .. ..$ : NULL
  .. ..$ : chr "6.3.0"
  .. ..$ : chr "6.3.0"
  .. ..$ : NULL
  .. ..$ : chr "6.3.0"
  .. ..$ : chr "6.3.0"
  .. ..$ : chr "6.3.0"
  .. ..$ : NULL
  .. ..$ : chr "6.3.0"
  .. ..$ : chr "6.3.0"
  .. ..$ : chr "6.3.0"
  .. ..$ : NULL
  .. ..$ : chr "6.3.0"
  .. ..$ : chr "6.3.0"
  .. ..$ : chr "6.3.0"
  .. ..$ : chr "6.3.0"
  .. ..$ : chr "6.3.0"
  .. ..$ : chr "6.3.0"
  .. ..$ : chr "6.3.0"
  .. ..$ : chr "6.3.0"
  .. ..$ : chr "6.3.0"
  .. ..$ : chr "6.3.0"
  .. ..$ : chr "6.3.0"
  .. ..$ : chr "6.3.0"
  .. ..$ : chr "6.3.0"
  .. ..$ : NULL
  .. ..$ : chr "6.3.0"
  .. ..$ : chr "6.3.0"
  .. ..$ : chr "6.3.0"
  .. ..$ : NULL
  .. ..$ : chr "6.3.0"
  .. ..$ : chr "6.3.0"
  .. ..$ : chr "6.3.0"
  .. ..$ : chr "6.3.0"
  .. ..$ : chr "6.3.0"
  .. ..$ : chr "6.3.0"
  .. ..$ : chr "6.3.0"
  .. ..$ : chr "6.3.0"
  .. ..$ : chr "6.3.0"
  .. ..$ : chr "6.3.0"
  .. ..$ : chr "6.3.0"
  .. ..$ : chr "6.3.0"
  .. ..$ : chr "6.3.0"
  .. ..$ : chr "6.3.0"
  .. ..$ : chr "6.3.0"
  .. ..$ : chr "6.3.0"
  .. ..$ : chr "6.3.0"
  .. ..$ : chr "6.3.0"
  .. ..$ : chr "6.3.0"
  .. ..$ : chr "6.3.0"
  .. ..$ : chr "6.3.0"
  .. ..$ : chr "6.3.0"
  .. ..$ : chr "6.3.0"
  .. ..$ : chr "6.3.0"
  .. ..$ : chr "6.3.0"
  .. ..$ : chr "6.3.0"
  .. ..$ : chr "6.3.0"
  .. ..$ : chr "6.3.0"
  .. ..$ : chr "6.3.0"
  ..$ package:List of 71
  .. ..$ : chr "dsBase"
  .. ..$ : chr "base"
  .. ..$ : chr "dsBase"
  .. ..$ : chr "dsBase"
  .. ..$ : chr "dsBase"
  .. ..$ : chr "dsBase"
  .. ..$ : chr "dsBase"
  .. ..$ : chr "dsBase"
  .. ..$ : chr "dsBase"
  .. ..$ : chr "dsBase"
  .. ..$ : chr "base"
  .. ..$ : chr "dsBase"
  .. ..$ : chr "stats"
  .. ..$ : chr "dsBase"
  .. ..$ : chr "dsBase"
  .. ..$ : chr "base"
  .. ..$ : chr "dsBase"
  .. ..$ : chr "dsBase"
  .. ..$ : chr "dsBase"
  .. ..$ : chr "base"
  .. ..$ : chr "dsBase"
  .. ..$ : chr "dsBase"
  .. ..$ : chr "dsBase"
  .. ..$ : chr "base"
  .. ..$ : chr "dsBase"
  .. ..$ : chr "dsBase"
  .. ..$ : chr "dsBase"
  .. ..$ : chr "dsBase"
  .. ..$ : chr "dsBase"
  .. ..$ : chr "dsBase"
  .. ..$ : chr "dsBase"
  .. ..$ : chr "dsBase"
  .. ..$ : chr "dsBase"
  .. ..$ : chr "dsBase"
  .. ..$ : chr "dsBase"
  .. ..$ : chr "dsBase"
  .. ..$ : chr "dsBase"
  .. ..$ : chr "base"
  .. ..$ : chr "dsBase"
  .. ..$ : chr "dsBase"
  .. ..$ : chr "dsBase"
  .. ..$ : chr "base"
  .. ..$ : chr "dsBase"
  .. ..$ : chr "dsBase"
  .. ..$ : chr "dsBase"
  .. ..$ : chr "dsBase"
  .. ..$ : chr "dsBase"
  .. ..$ : chr "dsBase"
  .. ..$ : chr "dsBase"
  .. ..$ : chr "dsBase"
  .. ..$ : chr "dsBase"
  .. ..$ : chr "dsBase"
  .. ..$ : chr "dsBase"
  .. ..$ : chr "dsBase"
  .. ..$ : chr "dsBase"
  .. ..$ : chr "dsBase"
  .. ..$ : chr "dsBase"
  .. ..$ : chr "dsBase"
  .. ..$ : chr "dsBase"
  .. ..$ : chr "dsBase"
  .. ..$ : chr "dsBase"
  .. ..$ : chr "dsBase"
  .. ..$ : chr "dsBase"
  .. ..$ : chr "dsBase"
  .. ..$ : chr "dsBase"
  .. ..$ : chr "dsBase"
  .. ..$ : chr "dsBase"
  .. ..$ : chr "dsBase"
  .. ..$ : chr "dsBase"
  .. ..$ : chr "dsBase"
  .. ..$ : chr "dsBase"
  ..$ type   : chr [1:71] "aggregate" "aggregate" "aggregate" "aggregate" ...
  ..$ class  : chr [1:71] "function" "function" "function" "function" ...
```

**Comment by timcadman** [comment]
Closed with PR #75
